internal/kafka/internal/presenters/kafka_test.go

```diff
@@ -37,12 +39,12 @@ func TestConvertKafkaRequest(t *testing.T) {
 				kafkaRequestPayload: *mock.BuildKafkaRequestPayload(nil),
 				dbKafkaRequests:     []*dbapi.KafkaRequest{},
 			},
-			want: mock.BuildKafkaRequest(
-				mock.With(mock.STATUS, ""),
-				mock.With(mock.OWNER, ""),
-				mock.With(mock.CLUSTER_ID, ""),
```

Comment:
Isn't 'empty string' the default? I believe you can remove the `mock.With` calls here.

Reply:
Not with `BuildKafkaRequest`, these fields are set now.

I did replace this one to use `BuildEmptyKafkaRequest` though which returns a kafka request object with no fields set so i've removed the mock.With calls here that sets these fields to an empty string. Reason for that is, the majority of the kafka requests expected by other tests requires most of the fields to be set. This particular test requires the majority of them to be empty. Rather than calling each mock.With to set these fields to an empty string, I just build an empty Kafka request and modify the fields to be set as what is expected for this test. :thinking: we don't really have a `BuildEmpty..` object for the rest of our mock objects yet. wdyt of doing it this way? :thinking:

Reply:
I commented the removed lines instead of the added ones... I guess I need some sleep... 

I would do it like this:

- `BuildKafkaRequest(Options options...)`: create an empty KafkaRequest that can be customised using the options

The I would add a `WithPredefinedTestValues` option that populates the request with the predefined values used by most tests.

This way, if you need the `pre populated` KafkaRequest you will do:
```go
kafkaRequest := BuildKafkaRequest(WithPredefinedTestValues())
````

WDYT?

Reply:
that sounds good :+1: i'll update that now.